### DISCUSSION TITLE
Start and end time as a planning variable.

### DISCUSSION THREAD

**Discussion by aayuwush** [TimefoldAI/timefold-solver#737] [General]
Can you make timefold add start and end time of a task based on its priority and time to complete the task? There will not be a fixed start and end time due to difference in completion time.  But to make a planning variable you need fixed start and end time like time slot. thank you!

**Comment by Christopher-Chianelli** [comment]
Consider using the chained through time pattern: https://docs.timefold.ai/timefold-solver/latest/design-patterns/design-patterns#chainedThroughTimePattern

![image](https://github.com/TimefoldAI/timefold-solver/assets/15825810/f9c93951-5485-44e2-a7d9-367cc5093fe0)

![image](https://github.com/TimefoldAI/timefold-solver/assets/15825810/510aba48-b2a8-4a55-80eb-4425a95a2087)

In summary, each "worker" has a time when they begin/depart to their first task, and the task's start and end time are shadow variables, depending on the previous task end time and current task start time respectively. See the [vehicle routing quickstart](https://github.com/TimefoldAI/timefold-quickstarts/tree/stable/use-cases/vehicle-routing) for an example.
